x/mongo/driver/topology/CMAP_spec_test.go

```diff
@@ -112,16 +126,16 @@ func runCMAPTest(t *testing.T, testFileName string) {
 		WithConnectionPoolMaxIdleTime(func(duration time.Duration) time.Duration {
 			return time.Duration(test.PoolOptions.MaxIdleTimeMS) * time.Millisecond
 		}),
+		WithConnectionPoolMaintainInterval(func(duration time.Duration) time.Duration {
+			return time.Duration(test.PoolOptions.BackgroundThreadIntervalMS) * time.Millisecond
```

Comment:
Ah, good find. AFAIK a negative value is not currently used because passing a negative duration to `time.NewTicker()` will cause a panic.

Do you think it's worth adding the logic to handle negative values now (i.e. call `WithConnectionPoolMaintainInterval` with a very large value) or just leave it as-is and handle the test panic when it comes up?

Reply:
Updated to disable the `maintain()` goroutine if given a negative `MaintainInterval`.

Reply:
I think this is resolved from the changes in 586f967c98656f564f04d9c843e3940da916bd94.